test/cctest/util.cc

```diff
@@ -132,3 +132,125 @@ TEST(UtilTest, UncheckedCalloc) {
+    for (size_t i = 0; i < old_length; i++)                                  \
+      EXPECT_EQ(static_cast<T>(i), buf[i]);                                  \
+    EXPECT_EQ(0, buf[old_length]);                                           \
+                                                                             \
+    /* SetLength() reduction and expansion */                                \
+    for (size_t i = 0; i < buf.length(); i++)                                \
+      buf[i] = static_cast<T>(i);                                            \
+    buf.SetLength(10);                                                       \
+    for (size_t i = 0; i < buf.length(); i++)                                \
+      EXPECT_EQ(static_cast<T>(i), buf[i]);                                  \
+    buf.SetLength(buf.capacity());                                           \
+    for (size_t i = 0; i < buf.length(); i++)                                \
+      EXPECT_EQ(static_cast<T>(i), buf[i]);                                  \
+                                                                             \
+    /* Subsequent Realloc */                                                 \
+    old_length = buf.length();                                               \
+    old_capacity = buf.capacity();                                           \
+    buf.AllocateSufficientStorage(old_capacity * 1.5);                       \
+    EXPECT_EQ(buf.capacity(), buf.length());                                 \
+    EXPECT_EQ(static_cast<size_t>(old_capacity * 1.5), buf.length());        \
+    EXPECT_TRUE(buf.IsAllocated());                                          \
+    for (size_t i = 0; i < old_length; i++)                                  \
+      EXPECT_EQ(static_cast<T>(i), buf[i]);                                  \
+                                                                             \
+    /* Basic I/O on Realloc'd buffer */                                      \
+    for (size_t i = 0; i < buf.length(); i++)                                \
+      buf[i] = static_cast<T>(i);                                            \
+    for (size_t i = 0; i < buf.length(); i++)                                \
+      EXPECT_EQ(static_cast<T>(i), buf[i]);                                  \
+                                                                             \
+    /* Release() */                                                          \
+    T* rawbuf = buf.out();                                                   \
+    buf.Release();                                                           \
+    EXPECT_EQ(0U, buf.length());                                             \
+    EXPECT_FALSE(buf.IsAllocated());                                         \
+    EXPECT_GT(buf.capacity(), buf.length());                                 \
+    free(rawbuf);                                                            \
+  } while (0)
```

Comment:
Consider putting this in a template function rather than a macro.  Rule of thumb: template when you can, macro when you must.